common/src/main/java/io/pravega/common/ConfigSetup.java

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.common;
+
+import io.pravega.common.util.ConfigBuilder;
+import java.util.function.Supplier;
+
+/**
+ * Interface used to pass the configuration to storage plugin implementation.
```

Comment:
This class is in `common`, yet you make references to `storage plugin` and `ServiceBuilder`. These terms do not belong here.

Can you:
- Find a new home for this
- Determine if this `ConfigSetup` is really needed. For instance, can we just pass in some lambda function wherever this is used instead of using this?

Reply:
- Moved to a different location. 
- This is sent to the factory which will be later used by anyone trying to create a storage binding. I think sending an interface is clearer.